src/Illuminate/Support/Lottery.php

```diff
@@ -0,0 +1,258 @@
+<?php
+
+namespace Illuminate\Support;
+
+use RuntimeException;
+
+class Lottery
+{
+    /**
+     * The number of expected wins.
+     *
+     * @var int|float
+     */
+    protected $chances;
+
+    /**
+     * The number of potential opportunities to win.
+     *
+     * @var int|null
+     */
+    protected $outOf;
+
+    /**
+     * The winning callback.
+     *
+     * @var null|callable
```

Comment:
Is it possible to annotate the callable here and for the looser so that phpstan knows what it accepts and returns? We might have to update the `choose` docblock then also to mind this.